staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -70,26 +76,31 @@ func ValidateAuthenticationConfiguration(c *api.AuthenticationConfiguration) fie
 	// check and add validation for duplicate issuers.
 	for i, a := range c.JWT {
 		fldPath := root.Index(i)
-		allErrs = append(allErrs, validateJWTAuthenticator(a, fldPath)...)
+		_, errs := validateJWTAuthenticator(a, fldPath, utilfeature.DefaultFeatureGate.Enabled(features.StructuredAuthenticationConfiguration))
+		allErrs = append(allErrs, errs...)
 	}
 
 	return allErrs
 }
 
 // ValidateJWTAuthenticator validates a given JWTAuthenticator.
 // This is exported for use in oidc package.
-func ValidateJWTAuthenticator(authenticator api.JWTAuthenticator) field.ErrorList {
-	return validateJWTAuthenticator(authenticator, nil)
+func ValidateJWTAuthenticator(authenticator api.JWTAuthenticator) (authenticationcel.CELMapper, field.ErrorList) {
+	return validateJWTAuthenticator(authenticator, nil, utilfeature.DefaultFeatureGate.Enabled(features.StructuredAuthenticationConfiguration))
 }
 
-func validateJWTAuthenticator(authenticator api.JWTAuthenticator, fldPath *field.Path) field.ErrorList {
+func validateJWTAuthenticator(authenticator api.JWTAuthenticator, fldPath *field.Path, structuredAuthnFeatureEnabled bool) (authenticationcel.CELMapper, field.ErrorList) {
 	var allErrs field.ErrorList
 
+	compiler := authenticationcel.NewCompiler(environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()))
+	mapper := &authenticationcel.CELMapper{}
```

Comment:
> if the feature's not enabled, probably don't need to construct these, right?

right now I'm passing the struct to different validate funcs that set the subfields instead of returning subfields, so we're initializing this. 

> hmm.. I can't tell if celMapper is conditional on the feature gate (will be nil if the feature's off) or on whether CEL expressions were used for various things (subfields will be nil if there's no CEL group expression, etc) or both.

In the current implementation, if the feature gate is not set, the subfields that contain an expression will be nil but celMapper is not nil. 

_We can return nil from `validateJWTAuthenticator` if feature gate is set._